### PR TITLE
Mention wasm-features in `Readme.md`

### PR DESCRIPTION
**Pull request by Glitchy-Tozier**
When important features aren't mentioned, this can lead to [some confusion](https://github.com/dariogoetz/keyboard_layout_optimizer/pull/7#issuecomment-1001672035)

Therefore, I suggest quickly mentioning them. :)
This pull-request adds a mention about WebAssembly and the required features.

---

Eventually, it would be handy to have every existing feature described in the `Readme.md`. Nothing extensive, just a short mention that the feature exists and an overview of what it does.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,6 +108,18 @@ BLAS backend in the `Cargo.toml`:
 ndarray-linalg = { version = "*", features = ["intel-mkl-static"] }
 ```
 
+#### Required for WebAssembly
```
Comment (stefan-k) [inline README.md:111]:
```suggestion
#### Experimental support for compiling to WebAssembly
```

File: README.md [hunk 2]
```diff
@@ -106,6 +108,18 @@ BLAS backend in the `Cargo.toml`:
...
+
+To prevent panics when compiling to WASM, one of the following features needs to be used:
```
Comment (stefan-k) [inline README.md:113]:
```suggestion
When compiling to WASM, one of the following features must be used:
```

File: README.md [hunk 3]
```diff
@@ -106,6 +108,18 @@ BLAS backend in the `Cargo.toml`:
...
+
+```toml
+argmin = { version = "0.4.7", features = ["wasm-bindgen"] }
+```
+
+```toml
+argmin = { version = "0.4.7", features = ["stdweb"] }
+```
+
```
Comment (stefan-k) [inline README.md:122]:
```suggestion
Note that WASM support is still experimental. Please report any issues you encounter when compiling argmin to WASM.

```

**Review by stefan-k — CHANGES_REQUESTED** [review]
Thank you for this PR and apologies for the confusion this has caused! 

The reason why this wasn't documented so far is that this feature isn't well tested and not part of the CI (see #89) since I don't need this feature myself and haven't had time yet to get into it in detail. I have suggested minor changes to your PR to reflect that. 

If you have experience with compiling to WASM, I would be very very happy if you could add a check to the CI as well :)

**Comment by Glitchy-Tozier** [comment]
I've implemented the requested changes! :)

> If you have experience with compiling to WASM, I would be very very happy if you could add a check to the CI as well :)

I don't have much experience, unfortunately. _Maybe_ @dariogoetz could share his sources with you though. He has set up the infrastructure of our keyboard-project.

**Comment by stefan-k** [comment]
Thanks for making the changes! I added the same text to `src/lib.rs` as I typically create the readme from there using `cargo readme`.

**Review by stefan-k — APPROVED** [review]

**Comment by Glitchy-Tozier** [comment]
By the way, I'm just curious: When do you plan on releasing `v0.5.0`?

**Comment by stefan-k** [comment]
I plan to release it until the end of January, hopefully sooner. But unfortunately I can't guarantee it ;)

**Comment by Glitchy-Tozier** [comment]
Understood  :)

**Comment by stefan-k** [comment]
v0.5.0 is released!

**Comment by Glitchy-Tozier** [comment]
Amazing, thank you for notifying me! :)

**Pull request merged**